contracts/contracts/strategies/BridgedWOETHStrategy.sol

```diff
@@ -0,0 +1,303 @@
+
+contract BridgedWOETHStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using StableMath for uint128;
+    using SafeCast for uint256;
+
+    event MaxPriceDiffBpsUpdated(uint128 oldValue, uint128 newValue);
+    event WOETHPriceUpdated(uint128 oldValue, uint128 newValue);
+
+    IWETH9 public immutable weth;
+    IERC20 public immutable bridgedWOETH;
+    IERC20 public immutable oethb;
+
+    uint256 public constant MAX_PRICE_STALENESS = 2 days;
+
+    uint128 public lastOraclePrice;
+    uint128 public maxPriceDiffBps;
+
+    /**
+     * @dev Verifies that the caller is the Governor or Strategist.
+     */
+    modifier onlyGovernorOrStrategist() {
+        require(
+            isGovernor() || msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist or Governor"
+        );
+        _;
+    }
+
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _weth,
+        address _bridgedWOETH,
+        address _oethb
+    ) InitializableAbstractStrategy(_stratConfig) {
+        weth = IWETH9(_weth);
+        bridgedWOETH = IERC20(_bridgedWOETH);
+        oethb = IERC20(_oethb);
+    }
+
+    function initialize(uint128 _maxPriceDiffBps)
+        external
+        onlyGovernor
+        initializer
+    {
+        InitializableAbstractStrategy._initialize(
+            new address[](0), // No reward tokens
+            new address[](0), // No assets
+            new address[](0) // No pTokens
+        );
+
+        _setMaxPriceDiffBps(_maxPriceDiffBps);
+    }
+
+    /**
+     * @dev Sets the max price diff bps for the wOETH value appreciation
+     * @param _maxPriceDiffBps Bps value, 10k == 100%
+     */
+    function _setMaxPriceDiffBps(uint128 _maxPriceDiffBps) internal {
+        require(
+            _maxPriceDiffBps > 0 && _maxPriceDiffBps <= 10000,
+            "Invalid bps value"
+        );
+
+        emit MaxPriceDiffBpsUpdated(maxPriceDiffBps, _maxPriceDiffBps);
+
+        maxPriceDiffBps = _maxPriceDiffBps;
+    }
+
+    /**
+     * @dev Sets the max price diff bps for the wOETH value appreciation
+     * @param _maxPriceDiffBps Bps value, 10k == 100%
+     */
+    function setMaxPriceDiffBps(uint128 _maxPriceDiffBps)
+        external
+        onlyGovernor
+    {
+        _setMaxPriceDiffBps(_maxPriceDiffBps);
+    }
+
+    /**
+     * @dev Finds the value of bridged wOETH from the Oracle.
+     *      Ensures that it's within the bounds and reasonable.
+     *      And stores it.
+     *
+     *      NOTE: Intentionally not caching `Vault.priceProvider` here,
+     *      since doing so would mean that we also have to update this
+     *      strategy every time there's a change in oracle router.
+     *      Besides on L2, the gas is considerably cheaper than mainnet.
+     *
+     * @return Latest price from oracle
+     */
+    function _updateWOETHOraclePrice() internal returns (uint256) {
```

Comment:
Throughout this file, when we have an external / internal pair of functions, like here, let's run the external version first, then the internal.